filebeat/tests/integration/filestream_test.go

```diff
@@ -105,3 +108,122 @@ func TestFilestreamCleanInactive(t *testing.T) {
 	registryFile := filepath.Join(filebeat.TempDir(), "data", "registry", "filebeat", "log.json")
 	filebeat.WaitFileContains(registryFile, `"op":"remove"`, time.Second)
 }
+
+func TestFilestreamValidationPreventsFilebeatStart(t *testing.T) {
+	duplicatedIDs := `
+filebeat.inputs:
+  - type: filestream
+    id: duplicated-id-1
+    enabled: true
+    paths:
+      - /tmp/*.log
+  - type: filestream
+    id: duplicated-id-1
+    enabled: true
+    paths:
+      - /var/log/*.log
+
+output.discard.enabled: true
+logging:
+  level: debug
+  metrics:
+    enabled: false
+`
+	emptyID := `
+filebeat.inputs:
+  - type: filestream
+    enabled: true
+    paths:
+      - /tmp/*.log
+  - type: filestream
+    id: unique-id-1
+    enabled: true
+    paths:
+      - /var/log/*.log
+
+output.discard.enabled: true
+logging:
+  level: debug
+  metrics:
+    enabled: false
+`
+	tcs := []struct {
+		name       string
+		cfg        string
+		wantErrLog string
+	}{
+		{
+			name:       "duplicated IDs",
+			cfg:        duplicatedIDs,
+			wantErrLog: "filestream inputs with duplicated IDs",
+		},
+		{
+			name:       "empty ID",
+			cfg:        emptyID,
+			wantErrLog: "input without ID",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			filebeat := integration.NewBeat(
+				t,
+				"filebeat",
+				"../../filebeat.test",
+			)
+
+			// Write configuration file and start Filebeat
+			filebeat.WriteConfigFile(tc.cfg)
+			filebeat.Start()
+
+			// Wait for error log
+			filebeat.WaitForLogs(
+				"filestream inputs validation error",
+				10*time.Second,
+				"Filebeat did log a validation error")
```

Comment:
should it be "Filebeat did not log a validation error"?
```suggestion
				"Filebeat did not log a validation error")
```